bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java

```diff
@@ -0,0 +1,125 @@
+package tools.vitruv.dsls.testutils;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.nio.file.Path;
+import static java.lang.System.lineSeparator;
+import static java.nio.file.Files.readString;
+import org.eclipse.jdt.core.compiler.CategorizedProblem;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.nio.file.Files.walk;
+import static com.google.common.collect.FluentIterable.from;
+
+import org.eclipse.xtext.util.JavaVersion;
+import org.eclipse.xtext.xbase.testing.InMemoryJavaCompiler;
+import org.eclipse.xtext.xbase.testing.InMemoryJavaCompiler.Result;
+import org.eclipse.xtext.xbase.testing.JavaSource;
+import static java.lang.reflect.Modifier.isPublic;
+import com.google.common.base.Predicate;
+
+import static org.eclipse.xtext.xbase.lib.IterableExtensions.join;
+
+import static com.google.common.base.Preconditions.checkState;
+
+public class InMemoryClassesCompiler {
```

Comment:
Could you add some documentation please? 🙄

Reply:
Done in ed5f36a.